Homework4/Taras_Samchuk_HW4_Task_1.cpp

```diff
@@ -0,0 +1,239 @@
+#include <chrono>
+#include <cstring>
+#include <fstream>
+#include <functional>
+#include <iostream>
+#include <random>
+
+enum SortType { ST_BOBLE_SORT, ST_SELECTION_SORT, ST_QUICK_SORT, COUNT_ST };
+
+using comparison = std::function<bool(int, int)>;
+
+const size_t k_int_array_size{10'000};
+const int k_min_random_value{10};
+const int k_max_random_value{99};
+const char k_file_name[]{"Taras_Samchuk_HW4_Task_1.txt"};
+
+bool More(int a, int b) { return a > b; }
+bool Less(int a, int b) { return a < b; }
+
+// RANDOM INIT ARRAY
+void rInitArray(int *const array_begin, int const *const array_end) {
+  if (array_begin == nullptr || array_end == nullptr ||
+      array_begin >= array_end) {
+    return;  // error
+  }
+  std::random_device rd;
+  std::mt19937 mt(rd());
+  std::uniform_int_distribution<int> GenInt(k_min_random_value,
+                                            k_max_random_value);
+  for (int *index = array_begin; index != array_end; index++) {
+    *index = GenInt(mt);
+  }
+  return;  // ok
+}
+
+void fInitArray(int *const array_begin, int const *const array_end) {
+  if (array_begin == nullptr || array_end == nullptr ||
+      array_begin >= array_end) {
+    return;  // error
+  }
+  std::ifstream InitFile;
+  InitFile.open(k_file_name);
```

Comment:
```suggestion
  std::ifstream InitFile{k_file_name};
```
можно открывать в конструкторе